addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -1705,3 +1705,193 @@ def test04_rereservation_for_same_batch(self):
+        return batch
+
+    def complete_pick(self, picking):
+        for move in picking.move_lines:
+            move.write({
+                'quantity_done': move.product_uom_qty,
+                'location_dest_id': self.test_output_location_01.id,
+            })
+
+    def test01_associates_pallet_with_initial_batch_picking(self):
+        batch = self.create_batch_with_picking()
+        picking = batch.picking_ids
+        batch.reserve_pallet('UDES11111', picking=picking)
+
+        self.assertEqual(picking.u_reserved_pallet, 'UDES11111')
+
+    def test02_associates_pallets_with_respective_pickings(self):
+        batch = self.create_batch_with_picking()
+        picking1 = batch.picking_ids
+        batch.reserve_pallet('UDES11111', picking=picking1)
+        picking2 = self.create_picking(self.picking_type_pick,
+                                       products_info=self.pack_4bananas_info,
+                                       confirm=True,
+                                       assign=True)
+        batch.add_extra_pickings(self.picking_type_pick.id)
+        batch.reserve_pallet('UDES11112', picking=picking2)
+
+        self.assertEqual(sorted(batch.mapped('picking_ids.u_reserved_pallet')), ['UDES11111', 'UDES11112'])
+
+    def test03_associates_pallet_with_added_extra_picking(self):
+        batch = self.create_batch_with_picking()
+        picking1 = batch.picking_ids
+        batch.reserve_pallet('UDES11111', picking=picking1)
+        picking2 = self.create_picking(self.picking_type_pick,
+                                       products_info=self.pack_4bananas_info,
+                                       confirm=True,
+                                       assign=True)
+        batch.add_extra_pickings(self.picking_type_pick.id)
+        batch.reserve_pallet('UDES11112', picking=picking2)
+
+        self.assertEqual(picking2.u_reserved_pallet, 'UDES11112')
+
+    def test04_raises_error_on_reservation_conflict(self):
+        batch1 = self.create_batch_with_picking()
+        picking1 = batch1.picking_ids
+        batch1.reserve_pallet('UDES11111', picking=picking1)
+        picking2 = self.create_picking(self.picking_type_pick,
+                                       products_info=self.pack_4bananas_info,
+                                       confirm=True,
+                                       assign=True)
+        batch1.add_extra_pickings(self.picking_type_pick.id)
+        batch1.reserve_pallet('UDES11112', picking=picking2)
+        batch2 = self.create_batch_with_picking()
+        picking3 = batch2.picking_ids
+
+        expected_error = 'This pallet is already being used for picking %s.' % \
+                         picking1.name
+
+        with self.assertRaisesRegex(ValidationError, expected_error,
+                                    msg='Incorrect error thrown'):
+            batch2.reserve_pallet('UDES11111', picking=picking3)
+
+    def test05_rereservation_for_same_batch(self):
```

Comment:
suggestion: test05_raises_error_on_reservation_conflict_same_batch

Reply:
Much better; I should apply my preferred naming convention (test_verb_condition) when copying tests from other classes :smile: